### PR TITLE
Mccalluc/add source for hca

### PR DESCRIPTION
**Pull request by mccalluc**
Follow-up to #628

- Actually add the special `source_project` column to `scrnaseq-hca.yaml`: It should have been added earlier.
- Move helper functions out of `submission`, mostly to make testing easier.

Filing as draft: I would like to add an end-to-end test.

@cebriggs7135 : 
- Can you provide a better description of `source_project`?
- For other assays, we are adding a required description... Should that be added for scrnaseq-hca as well?

### CODE REVIEW

**Review by ilan-gold — COMMENTED** [review]
Looks good, but would like to see the end-to-end.

**Comment by mccalluc** [comment]
@ilan-gold -- [Added end-to-end](https://github.com/hubmapconsortium/ingest-validation-tools/pull/673/files#diff-1ffd086240ac3b3529a06c264ab7e4a4b2a57ef71072c70b8a4484c215ac9081): Only fails because the contributors and data are not provided.

File: docs/scrnaseq-hca/v0.yaml [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Generated YAML: PRs should not start here!
 doc_url: https://portal.hubmapconsortium.org/docs/assays/rnaseq
 fields:
+- constraints:
+    enum:
+    - HCA
+    required: true
+  custom_constraints:
+    sequence_limit: 3
+  description: Source project
```
Comment (cebriggs7135) [inline docs/scrnaseq-hca/v0.yaml:10]:
```suggestion
  description: External source (outside of HuBMAP) of the project, eg. HCA (The Human Cell Atlas Consortium).
```

**Review by cebriggs7135 — COMMENTED** [review]
Added source description.

**Comment by cebriggs7135** [comment]
I added suggested text for the source_project description.

**Review by ilan-gold — APPROVED** [review]

**Comment by cebriggs7135** [comment]
@mccalluc I'll slack you an abbreviated (one dataset entry) HCA 10x metadata TSV along with a contributors.tsv and a fake fastq.gz file (you can put in a data_path directory) so you can create a complete submission directory to test your code on.

**Pull request merged**